doc/vim9.jax

```diff
@@ -0,0 +1,1298 @@
+	var mydict = {
+		one: 1,
+		two: 2,
+		}
+関数の呼び出し: >
+	var result = Func(
+			arg1,
+			arg2
+			)
+
+角カッコ []、波カッコ {}、または丸カッコの中‘以外で’二項演算子御使用する場
+合、その前後で改行することができます。例: >
+	var text = lead
+		   .. middle
+		   .. end
+	var total = start +
+		    end -
+		    correction
+	var result = positive
+			? PosFunc(arg)
+			: NegFunc(arg)
+
+"->" を使用したメソッド呼び出し、そしてドット (.) を使用したメンバー参照の場
+合、その前に改行を置くことができます: >
+	var result = GetBuilder()
+			->BuilderSetWidth(333)
+			->BuilderSetHeight(777)
+			->BuilderBuild()
+	var result = MyDict
+			.member
+
+<							*E1050*
+行頭の演算子と識別できるようにするために、範囲指定の前にはコロンを置きま
+す。"start" と "print" をつなげる例: >
+	var result = start
+	+ print
+これは以下の記述と同じです: >
+	var result = start + print
+
+次のように書くと、"start" を代入して、1行表示します: >
+	var result = start
+	:+ print
+
+Note |+cmd| の引数にはコロンは不要です: >
+	edit +6 fname
+
+関数の定義部においても、引数の間で改行をおくことができます: >
+	def MyFunc(
+		text: string,
+		separator = '-'
+		): string
+
+Notes:
+- "enddef" は継続行の先頭に置くことはできません。それは関数の終端を意味します。
+- 代入式の左辺を複数の行に分割できません。特にリストのアンパック |:let-unpack|
+  を使用する場合は注意が必要です。以下は問題のない例です: >
+	[var1, var2] =
+		Func()
+<  以下のように記述することはできません: >
+	[var1,
+	    var2] =
+		Func()
+- `:echo` や `:execute` のようなコマンドの引数は複数の行に分割できません。以下
+  は問題のない例です: >
+	echo [1,
+		2] [3,
+			4]
+<  以下のように記述することはできません: >
+	echo [1, 2]
+		[3, 4]
+- ラムダ関数の引数、"{" と "->" の間は複数の行に分割できません。以下は問題のな
+  い例です: >
+	filter(list, {k, v ->
+			v > 0})
+<  以下のように記述することはできません: >
+	filter(list, {k,
+			v -> v > 0})
+
+
+波括弧変数の廃止 ~
+
+波括弧変数 |curly-braces-names| は使用できません。
+
+
+辞書リテラル ~
+
+従来、Vim は波括弧 {} で辞書リテラルの表記をサポートしてきました: >
+	let dict = {'key': value}
+
+後に、辞書にシンプルなキーを使用することが非常に一般的であることが明らかになっ
+たため、キーをクォーテーションなしで指定できる表記が後方互換的に導入されまし
+た: >
+	let dict = #{key: value}
+
+しかし、この #{} という表記は他の言語に比べて異色なものです。キーには式よりも
+リテラルを使うほうが一般的で、JavaScript が使っている構文を考えると、辞書リテ
+ラルに {} の表記を使うほうがずっと便利です: >
+	let dict = {key: value}
+
+キーに式を使用する必要がある場合は、JavaScript と同様に角括弧を使用することが
+できます: >
+	let dict = {["key" .. nr]: value}
+
+
+:xit、:t、:append、:change、:insert の廃止 ~
+
+これらのコマンドは容易にローカル変数の名前と混同します。
+`:x` や `:xit` の代わりに `:exit` を使用できます。
+`:t` の代わりに `:copy` を使用できます。
+
+
+比較 ~
+
+オプション 'ignorecase' は文字列の比較には作用しません。
+
+
+スペース ~
+
+Vim9 script ではスペースの適切な使用を推奨しています。以下のような記述はできま
+せん: >
+	var name=234	# エラー!
+	var name= 234	# エラー!
+	var name =234	# エラー!
+"=" の前後にはスペースを置く必要があります: >
+	var name = 234	# OK
+コメント先頭の # の前にもスペースが必要です: >
+	var name = 234# エラー!
+	var name = 234 # OK
+
+多くの演算子の前後にはスペースが必要です。
+
+スペースをおいてはいけない場合:
+- 関数の名前と "(" の間: >
+	call Func (arg)	   # エラー!
+	call Func
+	     \ (arg)	   # エラー!
+	call Func(arg)	   # OK
+	call Func(
+	     \ arg)	   # OK
+	call Func(
+	     \ arg	   # OK
+	     \ )
+
+
+条件と式 ~
+
+条件と式は、他の言語とおよそ同じように扱われます。いくつかの値は旧来の Vim
+script と扱いが異なります:
+	value		legacy Vim script	Vim9 script ~
+	0		falsy			falsy
+	1		truthy			truthy
+	99		truthy			Error!
+	"0"		falsy			Error!
+	"99"		truthy			Error!
+	"text"		falsy			Error!
+
+"??" 演算子か "!" を使用している場合はエラーとなることはなく、すべての値は真ま
+たは偽として評価されます。真偽の評価方法は JavaScripot とほぼ同じですが、空の
+リストと辞書は偽として評価されます:
```

Comment:
eval.{txt,jax} の *falsy* *truthy* の節を見てもらって、ちょっと訳を調整したほうがよさそうかとは思いました。
ここは私も h-east さんと悩んだところなので、これが正解という指摘はできないのですが...

Reply:
```diff
 "??" 演算子か "!" を使用している場合はエラーとなることはなく、すべての値は真ま
-たは偽として評価されます。真偽の評価方法は JavaScripot とほぼ同じですが、空の
-リストと辞書は偽として評価されます:
+たは偽として評価されます。これは JavaScripot とほぼ同じですが、空のリストと辞
+書は偽として評価されます:
```

参照しましたが、ちょっと引っかかりポイントが分からなかったです

Reply:
すでに対応ははいってるので、以下はまあ参考までなのですが

たとえば *falsy* で

```
The expression before the '??' is evaluated.  If it evaluates to
|truthy|, this is used as the result.  Otherwise the expression after the '??'
is evaluated and used as the result.  This is most useful to have a default
value for an expression that may result in zero or empty: >
```

を

```
'??' の前の式が評価される。|truthy| と評価された場合、これが結果として使われ
る。
そうでないなら、'??' の後の式が評価され、結果として使われる。これがもっとも便
利なのは、ゼロや空な結果になりうる式でデフォルト値を持つ場合: >
```

としてまして。

falsyを偽とだけにしてしまうと、falseとの違いなどもありどうしたものかなと

```
For the "??" operator and when using "!" then there is no error, every value
is either falsy or truthy.  This is mostly like JavaScript, except that an
empty list and dict is falsy:
```

を

```
"??" 演算子か "!" を使用している場合はエラーとなることはなく、すべての値は真ま
たは偽として評価されます。真偽の評価方法は JavaScripot とほぼ同じですが、空の
リストと辞書は偽として評価されます:
```

としてますが、falsyのままにして

```
"??" 演算子か "!" を使用している場合はエラーとなることはなく、すべての値は falsy
か truthy として評価されます。評価方法は JavaScripot とほぼ同じですが、空のリス
トと辞書は falsy として評価されます:
```

など... とか?

---

see [404 Blog Not Found:"\-y"は「っぽい」って訳していいっぽい](http://blog.livedoor.jp/dankogai/archives/51704592.html)
and PR comment https://github.com/vim-jp/vimdoc-ja-working/pull/862#discussion_r534118823

Reply:
なるほど...悩ましいですねこれ。普通にtruthy / falsyが良さそうですね。